File: src/main/java/com/sudosoo/takeiteasy/dto/category/CategoryResponseDto.java
package com.sudosoo.takeiteasy.dto.category;

import com.sudosoo.takeiteasy.dto.post.PostTitleOnlyResponseDto;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.domain.Page;

@Getter
@NoArgsConstructor(access = AccessLevel.PRIVATE)
public class CategoryResponseDto {

    private String categoryName;
    private Page<PostTitleOnlyResponseDto> postResponseDtos;

    public CategoryResponseDto(String categoryName, Page<PostTitleOnlyResponseDto> postTitleDtos) {
        this.categoryName = categoryName;
        this.postResponseDtos = postTitleDtos;
    }
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/dto/category/CategoryResponseDto.kt
package com.sudosoo.takeItEasy.dto.category

import com.sudosoo.takeItEasy.dto.post.PostTitleOnlyResponseDto
import org.springframework.data.domain.Page

class CategoryResponseDto(
    val categoryName: String,
    val postTitleDtos: Page<PostTitleOnlyResponseDto>
)